nemo/core/connectors/save_restore_connector.py

```diff
@@ -454,8 +488,25 @@ def _handle_artifacts(self, model, nemo_file_folder):
                 # change back working directory
                 os.chdir(cwd)
 
+    def _update_subconfigs(self, model, path2yaml_file):
```

Comment:
Do we really need nested configs? Can you make the name closer to _update_subconfig_for_artifacts if you think this wont be used anywhere else? + move this function to SaveRestoreConnector maybe?

Reply:
> Do we really need nested configs?
Yes. The problem is that the child model can change its config (not only artifacts! e.g. you can change number of heads, momentum parameter in BatchNorm and so on). There is no easy way to notify parent about these config changes (some hacks exist, but it will be too complicated and fragile). So when saving we have to update subconfigs.
There is a test for this: see `change_child_number` parametrization (child model changes config, this should be handled correctly).

>  move this function to SaveRestoreConnector maybe?
Didn't understand, it is in `SaveRestoreConnector`. 

I marked it with `@staticmethod`, since it is really static.